### ISSUE TITLE
Occasionally alarm doesn't fire

### ISSUE THREAD

**Issue by nu111** [yuriykulikov/AlarmClock#468]
A couple of times the alarm setted for 7:00 for working days didn't ring.

Unfortunately I don't know how to reproduce this.

After the lost alarm, a popup appeared "alarm deactivated after 10 minutes"
![Screenshot_20220622-070818](https://user-images.githubusercontent.com/1246989/174954734-58dc6f39-e1b9-4ee5-ab14-8d03ecaebcd8.png)

App version: 3.10.10
Device: Google Pixel 4a
Android 12

**Comment by yuriykulikov** [comment]
This most likely means that do not disturb mode doesn't have an exception
for alarms. Please check do not disturb mode settings.

**Comment by nu111** [comment]
I always use "do not disturb" mode while I sleep. Normally it does not prevent alarm to ring.

**Comment by yuriykulikov** [comment]
Please check if alarms are allowed to interrupt

**Comment by nu111** [comment]
Hi, thanks for your help.

This setting is on `welfare >  do not disturb`:
![screen](https://user-images.githubusercontent.com/1246989/175535222-9f787bd5-3ee3-42cb-8b0c-af26d38d71a5.png)

I see another interesting one `App > App special access > Do not disturb access`
your app isn't in the list, but I don't have a button to add apps to the list.

**Comment by yuriykulikov** [comment]
In this case what remains is to check the volume. You can test the volume
in the application settings. Moving the slider should make some sound.

**Comment by nu111** [comment]
Yep! It sounds. I must say that even the "native" Google alarm sometimes had the same behavior, but much more frequent. I've choose your app cause I've read that is more reliable. Never had this issue, it happened 2 times in (I think) the last 2 months

**Comment by yuriykulikov** [comment]
This shouldn't happen on Pixel devices. I am quite confused about that.

**Comment by ronso0** [comment]
I can confirm it occasionally doesn't fire the alarm, just the notification shows up.

I noticed this twice and also suspected it was due to Do Not Disturb mode. (even though the working and not working alarms were set to aboit the same time in the morning, both within my DND time)
So I tested with an alarm scheduled for 2min later, and it **did** go off.
Then I adjusted the time for the morning after, and it didn't go off, just the (silent) notification.

Other alarm apps work just fine under exact same circumstances.
 I've chosen it because I loke the UI, specifically that I can read the alarm label when the screen is very dim, but it's unusable for me atm, it needs to be reliable.

This is version 3.15.01 from FDroid
Zenfone8
Android 13

**Comment by ronso0** [comment]
Sorry, what I described above appearantly (now) also occurs with other alarm apps (alarm doesn't go off until I wake the display).
I just noticed the default clock app was disabled and another one with an almost identical app icon was in use (might have been the Google clock, I didn't verify before uninstalling that).

I'll give AlarmClock anothe shot, once I'm sure the system alarm works reliably again.
